index.js

```diff
@@ -34,14 +32,10 @@ function createRouter (routes) {
 
   return function (store) {
     store.on('@init', function () {
-      store.dispatch(change, parse(loc.pathname, routes))
+      store.dispatch(change, parse('/', routes))
```

Comment:
Because there is no way to determine what url is requested from the server, but store should be initialized ith initial state.
User should dispatch store manually with current request url.

Reply:
The same component will be always rendered. Then what profit will we get with this approach?

Reply:
You think that there is no need to initialize state on the server by default?

Reply:
How Storeon will render requested page (if users asked for `/posts/1` we need to route this URL to render this page)

Reply:
Yes. You need to show the desired page.

Reply:
@ai `navigate` should be dispatched for any requested page (if I understand the question correctly). Default state initialization already removed after discuss with @isolovev

Reply:
Can you show an example of how to set the path for the router in SSR?

Reply:
`store.dispatch(router.navigate, "/my/great/url/")`

Reply:
It will be slower than just setting `window.location = '/my/path'` before creating Storeon.

If your case app will render home page first and only then will render the right page.

Reply:
> It will be slower than just setting window.location = '/my/path' before creating Storeon.

In most ssr cases store creation declared once and then imported into `client` and `server` entrypoint.

> If your case app will render home page first and only then will render the right page.

No, beacuse store passed to component before server rendering.

Reply:
And I think working with `window` object anywhere in server-specific code is not okay. It maybe okay in library, but not in app.

Reply:
Давай по-русски, я что-то запутался.

Вот node.js сервер инициарует стор, там `/`. На этот путь запускаются какие-то вычисления в сторе (например загрузка постов). А только потом мы меняет путь на другой и передаём стор в рендер.

Тем более разработчики могут сделать ошибку и работать с SSR так: иницируем стор, рендерим компонент, меняем путь.

Reply:
> And I think working with window object anywhere in server-specific code is not okay. It maybe okay in library, but not in app.

Куча сторонних библиотек в реальном приложении всё равно будут требовать `window`. Да и очень легко забыть об SSR и написать `window` в коде приложения.

Reply:
> Вот node.js сервер инициарует стор, там /. На этот путь запускаются какие-то вычисления в сторе (например загрузка постов). А только потом мы меняет путь на другой и передаём стор в рендер.

Где-то здесь мы решили, что инициализировать стор с `/` на сервере не ок. Нужно обязательно сделать dispatch с запрошенным пользователем адресом.

> Тем более разработчики могут сделать ошибку и работать с SSR так: иницируем стор, рендерим компонент, меняем путь.

Тогда ничего не будет работать совсем, потому что результат серверного рендеринга - строка. Можно что угодно делать со стором после, она никогда не изменится.

Reply:
> Да, круто когда библиотека может работать без window, но это круто если для этой работы не надо жервтвоать чем-то большим. Пока мы жертвуем поддериваемостью кода и сменой URL в процессе работы.

Придумывать какие-то "обходные решения" в своем приложении в таких случаях ок (если библиотеку очень хочется), но мне кажется хорошо заставлять пользователя их использовать. Например, если сделать так, то это будет пока что единственная библиотека на сервере, которая потребует от меня `window`

Reply:
Как я вижу эту проблему — у нас есть практически проблемы (дублирование кода, двойная обработка) и вопрос чистоты (не надо завязываться на `window`).

Я согласен, что вопрос чистоты всегда важен. Но он менее важен, чем практические проблемы. Мы не должны ради чистоты делать жить пользователей сложнее. Если бы не было практических проблем, то я бы с тобой согласился.

Я согласен, что использовать `window` — не круто. Но пока это выглядит меньшим злом.

Reply:
Мне кажется вполне ок вариант с моками, потому что это будет спрятано в библиотеке. Меня смущает только код в @init - совсем-совсем не хочется в коде сервера работать с window. В качестве решения можно location делать с "/" по умолчанию.

Ещё я переживаю, что какая-то библиотека может по наличию window определять, на сервере она или нет - тогда мы можем сломать работу совсем другого кода. Или мы можем как-то изолировать моки внутри библиотеки?

Reply:
Впрочем, у меня есть идеи как решить последнее.

Reply:
> В качестве решения можно location делать с "/" по умолчанию.

Ну мы как раз обсуждали, что это не круто, так как из-за `/` могут запуститься каакие-то модули

> Ещё я переживаю, что какая-то библиотека может по наличию window определять, на сервере она или нет - тогда мы можем сломать работу совсем другого кода. 

Такие библиотеки не будут работать с большинством SSR решений. Тот же Preact делает SSR вообще через Puppeter.

Reply:
> Ну мы как раз обсуждали, что это не круто, так как из-за / могут запуститься каакие-то модули

Если так не делать, то придется в коде сервера приложения и правда обращаться к `windowё. Чувство прекрасного мне не позволит так делать и я не стану этим пользоваться.
Если других вариантов решений нет (у меня - нет), то можем закрывать этот PR. Возможно, однажды кто-то придет и придумает как это сделать лучше.

Reply:
Попробуй https://github.com/majo44/storeon-async-router — там и фич больше и мейнтейнер может пойдёт на другой компромисс между чувством прекрасного и реальных проблем.